### PR TITLE
Enable faster cuBLAS path for torch.linalg.lstsq for batch of small matrices

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR enables cuBLAS path for `torch.linalg.lstsq`. Before this PR only cuSOLVER path was used for regular PyTorch builds (when built with MAGMA).

Performance results (also previously reported at https://github.com/pytorch/pytorch/pull/54725#issuecomment-832234456):
```
|                            | before current PR | current PR | speedup |
|----------------------------|-------------------|------------|---------|
| torch.Size([32, 32, 32])   | 870               | 440        | 2x      |
| torch.Size([64, 32, 32])   | 1340              | 450        | 3x      |
| torch.Size([32, 64, 64])   | 9040              | 1839       | 5x      |
| torch.Size([64, 64, 64])   | 17000             | 1830       | 9.2x    |
| torch.Size([32, 128, 128]) | 23210             | 8560       | 2.7x    |
| torch.Size([64, 128, 128]) | 40000             | 8662       | 4.6x    |
| torch.Size([32, 256, 256]) | 58160             | 46150      | 1.2x    |
| torch.Size([64, 256, 256]) | 73220             | 52080      | 1.4x    |
Times are in microseconds (us).
```

### CODE REVIEW

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp [hunk 1]
```diff
@@ -3231,27 +3231,20 @@ void lstsq_kernel(const Tensor& a, Tensor& b, Tensor& /*rank*/, Tensor& /*singul
...
       gels_looped(a, b, infos);
     }
 #else
-    // if both MAGMA and cuSOLVER are available this would call cuSOLVER
-    // MAGMA is called if cuSOLVER is not available
-    gels_looped(a, b, infos);
-#endif // AT_MAGMA_ENABLED()
+    // On ROCm platform we can only use MAGMA here
+    // If MAGMA is not available, an error will be thrown
+    gels_magma(a, b, infos);
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp:3246]: Why change the behavior on ROCm?

**Review by mruberry — APPROVED** [review]
Cool! -- One question for you, @IvanYashchuk

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp (hunk 1 above)
Comment (IvanYashchuk) [reply aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp:3246]: It doesn't change it. It makes it more explicit that on ROCm magma function is called. Previously the chain of calls was "this_function() -> gels_looped() -> gels_magma()", now it's "this_function() -> gels_magma()".

Comment (mruberry) [reply aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp:3246]: Cool, thanks for the explanation

**Pull request closed without merging**